File: services/FlightBoard/Domain/Entities/Flight.cs
namespace Domain.Entities;

public class Flight
{
    public int FlightId { get; set; }
    public int AirlineId { get; set; }
    public Airline Airline { get; set; } = null!;
    public int AircraftId { get; set; }
    public Aircraft Aircraft { get; set; } = null!;
    public string Number { get; set; } = null!;
    public int SourceId { get; set; }
    public Airport Source { get; set; } = null!;
    public int DestinationId { get; set; }
    public Airport Destination { get; set; } = null!;
    public FlightStatus Status { get; set; }
}

File: projects/flight-board-ui/projects/flight-board/src/app/flights/state/models/flight.ts
import { FlightStatus } from './flight-status';

export interface Flight {
    flightId: number;
    flightNumber: string;
    airlineName: string;
    destination: string;
    scheduledDepartureTime: Date;
    estimatedDepartureTime: Date;
    actualDepartureTime: Date;
    status: FlightStatus;
    departureGate: string;
}
